src/coreComponents/managers/Functions/TableFunction.hpp

```diff
@@ -177,16 +297,170 @@ class TableFunction : public FunctionBase
 
   /**
    * @brief The corners of the box that surround the value in N dimensions
-   * m_corners should be of size m_maxDimensions x (2^m_maxDimensions)
+   * m_corners should be of size maxDimensions x (2^maxDimensions)
    */
-  localIndex m_corners[m_maxDimensions][16];
+  localIndex m_corners[maxDimensions][16];
 
   /// The number of active table corners
   localIndex m_numCorners;
+
+  /// Kernel wrapper to interpolate in table and return derivatives
+  KernelWrapper m_kernelWrapper;
+
 };
 
-ENUM_STRINGS( TableFunction::InterpolationType, "linear", "nearest", "upper", "lower" )
+template< typename IN_ARRAY, typename OUT_ARRAY >
+GEOSX_HOST_DEVICE
+void
+TableFunction::KernelWrapper::compute( IN_ARRAY const & input, real64 & value, OUT_ARRAY && derivatives ) const
+{
+  value = 0.0;
+  for( localIndex i = 0; i < m_dimensions; ++i )
+  {
+    derivatives[i] = 0.0;
+  }
 
+  // Linear interpolation
+  if( m_interpolationMethod == TableFunction::InterpolationType::Linear )
```

Comment:
@TotoGaz in the current version of the PR, I don't touch this file anymore, so I will leave this comment for a subsequent PR. There is a lot of room for improvement in the `compute` function of `TableFunction` for use in the CO2-brine model. In particular, since we construct internally the table coordinates with uniform spacing, it does not make much sense to use binary search. Will be addressed in another PR.